### ISSUE TITLE
Add internal Aliases for Library and Examples

### ISSUE THREAD

**Issue by KiwiKilian** [maplibre/maplibre-react-native#515]
I would propose:

- Library: `@/*` for `./src/*`
- Examples: `@examples/*` for `./packages/examples/*`

Opinions @tyrauber?

**Comment by tyrauber** [comment]
That is what I would do.

**Comment by KiwiKilian** [comment]
It works easily within the example app, but it's difficult for the library, because the aliases have to be resolved during build, which is not quite what [tsc aliases are made for](https://github.com/callstack/react-native-builder-bob/issues/56#issuecomment-640264604).

**Comment by brentforder** [comment]
Can we handle this at build time in the tsconfig as so?
```
"compilerOptions": {
  ...
  "paths": {
      "@/*": ["src/*"],
      "@examples/*": ["packages/examples/*"]
    },
},

```
That's how I'm doing import path aliasing in my app.

**Comment by KiwiKilian** [comment]
Yes that works great for apps, but not for libraries, see the linked comment in https://github.com/maplibre/maplibre-react-native/issues/515#issuecomment-2525716598. Therefore I'm thinking to only add the aliases as follows:

- For Jest Tests:
  - `@maplibre/maplibre-react-native` for public exports
  - `@/**/*` for internal exports
- Within Examples: `@/*` for `./packages/examples/shared/src/*`

**Comment by tyrauber** [comment]
Thanks @KiwiKilian, this is awesome!
